proto/identity/associations/association.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log, which is considered valid iff the XID can be
+// deterministically derived from the address and nonce.
+// The recovery address defaults to the initial_associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  string initial_associated_address = 3;
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+    ASSOCIATED_ROLE_LEGACY_KEY = 3;
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
```

Comment:
What is the `client_timestamp_ns` used for? Seems dangerous to trust it for anything other than using it as a nonce.

Reply:
Honestly, I think both server timestamps and client timestamps are untrustworthy, depending on the trust model. I think it is helpful to assume that any consumer will account for this, and to include both and explicitly label where they came from.

The uses for the client timestamp here could be: as a nonce, as a human readable value in the signature that the user can verify, as a piece of metadata that the user can use to match an installation to an app they have installed.